src/main/java/org/reaktivity/nukleus/sse/internal/stream/ServerStreamFactory.java

```diff
@@ -814,6 +836,73 @@ private void handleReset(
             final long traceId = reset.trace();
             doReset(applicationReplyThrottle, applicationRouteId, applicationReplyId, traceId);
         }
+
+        private void handleChallenge(
+            ChallengeFW challenge)
+        {
+            final HttpChallengeExFW httpChallengeEx = challenge.extension().get(httpChallengeExRO::tryWrap);
+            if (httpChallengeEx != null)
+            {
+                final JsonObject jsonHeaders = new JsonObject();
+                final JsonObject challengeJson = new JsonObject();
+                final ListFW<HttpHeaderFW> httpHeaders = httpChallengeEx.headers();
+
+                httpHeaders.forEach(header ->
+                {
+                    final String name = header.name().asString();
+                    final String value = header.value().asString();
+                    if (name != null)
+                    {
+                        if (!name.startsWith(":"))
+                        {
+                            jsonHeaders.addProperty(name, value);
+                        }
+                        else if (name.equals(":method"))
+                        {
+                            challengeJson.addProperty("method", value);
+                        }
+                    }
+                });
+                challengeJson.add("headers", jsonHeaders);
+                challengeBuffer.putStringUtf8(0, gson.toJson(challengeJson));
+                final StringFW challengeData = stringRO.wrap(challengeBuffer, 0, challengeBuffer.capacity());
+
+                final SseEventFW sseEvent = sseEventRW.wrap(challengeBuffer, DataFW.FIELD_OFFSET_PAYLOAD, challengeBuffer.capacity())
+                        .type(challengeEventType.value())
+                        .data(challengeData)
+                        .build();
+
+                int applicationReplyPadding = networkReplyPadding + MAXIMUM_HEADER_SIZE;
+                final int applicationReplyCredit = networkReplyBudget - applicationReplyBudget;
+                if (applicationReplyCredit > 0)
+                {
+//                    final long traceId = window.trace();
+//                    final long authorization = window.authorization();
+//                    doWindow(applicationReplyThrottle, applicationRouteId, applicationReplyId, traceId, authorization,
+//                            applicationReplyCredit, applicationReplyPadding, window.groupId());
+
+                    applicationReplyBudget += applicationReplyCredit;
+                }
+
```

Comment:
Next step is to check there is enough `networkReplyBudget` to support sending the `DataFW` payload with `networkReplyPadding`.